### ISSUE TITLE
FYI only:  too many nonzero values of e00300 in the puf, but weighted sum is not far off

### ISSUE THREAD

**Issue by donboyd5** [PSLmodels/Tax-Calculator#2505]
Not a major issue but possibly worth thinking about as you think about weighting.

I looked at taxable interest (e00300) in the puf advanced to 2017 with stages 1 and 2, filers only as defined in #2501, in comparison to the comparable IRS aggregate from Table 1.4, screenshot below.

![image](https://user-images.githubusercontent.com/6538392/97808169-56cb3580-1c33-11eb-897d-fd089f6afe0a.png)

The weighted sum of e00300 in the 2017 puf is 4.3% above the IRS value, but the weighted number of filers with nonzero e00300 is 36.4% greater. I didn't use stage 3 and if I had it would certainly produce a different distribution of e00300 across agi ranges than in the data I have now, but I doubt it would have much impact on the top-line numbers I mention here.

I looked at the weighted distribution of nonzero values of e00300 in the 2017 puf (that is, the univariate distribution, not the distribution by agi range), and 50% of the weighted records with nonzero values had taxable interest < $98. Given that the values are so small, I think the far-too-many number of nonzero values is unlikely to have any meaningful impact on tax policy analyses. For my immediate purposes, I will only target the weighted sum of e00300 and not the weighted number of records with nonzero values, as targeting the number would probably force the data to do gyrations to try to hit the IRS number of nonzero values.

I don't think it's a huge issue but it might bear investigation at some point into why so many nonzero e00300 values have crept into the puf data.

**Comment by andersonfrailey** [comment]
Thanks for reporting this, @donboyd5. Can you look where those units with a low value for e00300 are coming from the PUF or CPS? You can use the `data_source` variable to do it. It'll be 1 if the record is based on a PUF unit, 0 if it's from the CPS.

**Comment by donboyd5** [comment]
Thanks, @andersonfrailey.

The national IRS data show approximately 44 million returns with nonzero e00300 in 2017 (see above) but my version of puf 2017 has about 60 million, so we (or at least I - perhaps I did something wrong) have about 16 million too many.

Here is a table with weighted counts of records by size of e00300 taxable interest, for a 2017 puf, giving a breakdown of the 60 million by data_source and size of e00300. (It only includes those I define as filers (#2501), and excludes anyone who had exactly zero e00300. Also, the puf is different from what comes straight out of tax-calculator because I did not do stage 3, which of course affects e00300 directly. I do not think that would have much if any influence on the number of nonzero e00300 records.)

![image](https://user-images.githubusercontent.com/6538392/97895578-f656f980-1d01-11eb-811a-e66eeb8666db.png)

While a disproportionate share of the very-low-value nonzero e00300 records do come from the cps (including all of the <$1 nonzero e00300 puf records), (a) I suppose that's not a surprise, as we are filling in a lot of lower-income returns from the cps and presumably they wouldn't have large e00300, (b) that probably wouldn't be a concern to us as <$1 e00300 couldn't possibly affect tax policy analysis in a meaningful way, and (c) the cps-sourced records apparently can't be the major factor behind the issue as there are only 4.5 million cps-based weighted records with nonzero e00300 and we have a discrepancy vs. the IRS of 16 million weighted records).

So it seems to me like the major source of the issue is the puf itself. I'll go back and quadruple-check what I've done but I don't think it's anything I've done wrong. Maybe it's been an issue in the puf forever - just one way that the puf is different from the data the IRS uses for official statistics, perhaps due to blurring or other things they did. I suppose I could look at puf2011.csv to see if that's the case. I don't know taxdata and tax-calculator well enough to know if either imputes interest income to records that don't have it in the source data but maybe something is there.

I'll try to look a little more. I'm guessing it's not an issue for analytic purposes as there are a lot of small-dollar amounts but it would be nice to understand.
